### PR TITLE
CP-18612: Introduce multiple flow tables for pvs vswitch rules

### PR DESCRIPTION
**Pull request by anoobs**

### CODE REVIEW

**Comment by anoobs** [comment]
@robhoes, can you check ?

File: scripts/vif-real [hunk 1]
```diff
@@ -188,9 +188,22 @@ add_to_bridge()
         fi
 
         $vsctl --timeout=30 -- --if-exists del-port $dev -- add-port $bridge $dev $vif_details || handle_error "Failed to vsctl add-port ${bridge} ${dev} ${vif_details}"
+        ofport=$($vsctl get interface "$dev" ofport)
+        if [ $? -ne 0 ] || [ -z "$ofport" ]; then
+            handle_error "$dev interface not foind on a bridge"
```
Comment (robhoes) [inline scripts/vif-real]: "(...) not found on the bridge"

File: scripts/vif-real [hunk 2]
```diff
@@ -188,9 +188,22 @@ add_to_bridge()
...
+        fi
+
+        xenstore-write "${PRIVATE}/"$TYPE"-ofport" "$ofport"
```
Comment (robhoes) [inline scripts/vif-real]: Quotation looks wrong.

File: scripts/vif-real [hunk 3]
```diff
@@ -227,8 +240,11 @@ remove_from_bridge()
         # port.  Use --if-exists to suppress the error that would otherwise
         # arise in that case.
         $vsctl --timeout=30 -- --if-exists del-port $dev
+        xenstore-rm ${PRIVATE}/"$TYPE"-ofport
```
Comment (robhoes) [inline scripts/vif-real]: No need to clean up; xenopsd will do it.

File: scripts/vif-real [hunk 4]
```diff
@@ -188,9 +188,22 @@ add_to_bridge()
...
+        if [ $? -ne 0 ] || [ -z "$ofport" ]; then
+            handle_error "$dev interface not foind on a bridge"
+        fi
+
+        xenstore-write "${PRIVATE}/"$TYPE"-ofport" "$ofport"
         ;;
     esac
 
+    ifindex=$(cat /sys/class/net/"$dev"/ifindex)
+    if [ $? -ne 0 ] || [ -z "$ifindex" ]; then
```
Comment (robhoes) [inline scripts/vif-real]: Is `$?` meaningful here?

File: scripts/vif-real (hunk 2 above)
Comment (anoobs) [inline scripts/vif-real]: I will fix it.

File: scripts/vif-real (hunk 1 above)
Comment (anoobs) [inline scripts/vif-real]: Will fix it.

File: scripts/vif-real (hunk 3 above)
Comment (anoobs) [inline scripts/vif-real]: sure. i will remove them.

Comment (anoobs) [inline scripts/vif-real]: On a second thought, I added xenstore-rm here, to remove tap-ofport and tap-ifindex (once pv driver kicks in). As this is shared between vif/tap, tap-* will linger till the whole path is removed.

**Comment by robhoes** [comment]
Looks good now. Just waiting for @anoobs to do a bit more testing before we put this in.

**Comment by anoobs** [comment]
I have done some testing with multiple IP addresses and multiple sites, they look good. This PR can be merged now.

**Pull request merged**